samples/sqlservices/src/controllers/mainController.ts

```diff
@@ -40,11 +40,110 @@ export default class MainController implements vscode.Disposable {
 			vscode.window.showInformationMessage(`Clicked from profile ${profile.serverName}.${profile.databaseName}`);
 		});
 
+		vscode.commands.registerCommand('mssql.openDialog', () =>  {
+			this.openDialog();
+		});
+
 		return Promise.resolve(true);
 	}
 
+	private openDialog(): void {
+		let dialog = sqlops.window.modelviewdialog.createDialog('Test dialog');
+		let tab1 = sqlops.window.modelviewdialog.createTab('Test tab 1');
+
+		let tab2 = sqlops.window.modelviewdialog.createTab('Test tab 2');
+		tab2.content = 'sqlservices';
+		dialog.content = [tab1, tab2];
+		dialog.okButton.onClick(() => console.log('ok clicked!'));
+		dialog.cancelButton.onClick(() => console.log('cancel clicked!'));
+		dialog.okButton.label = 'ok';
+		dialog.cancelButton.label = 'no';
+		let customButton1 = sqlops.window.modelviewdialog.createButton('Test button 1');
+		customButton1.onClick(() => console.log('button 1 clicked!'));
+		let customButton2 = sqlops.window.modelviewdialog.createButton('Test button 2');
+		customButton2.onClick(() => console.log('button 2 clicked!'));
+		dialog.customButtons = [customButton1, customButton2];
+		tab1.registerContent(async (view) => {
+			let inputBox = view.modelBuilder.inputBox()
+			.withProperties({
+				//width: 300
+			})
+			.component();
+			let inputBox2 = view.modelBuilder.inputBox()
+			.component();
+
+			let inputBox3 = view.modelBuilder.inputBox()
+			.component();
+
+			let checkbox = view.modelBuilder.checkBox()
+			.withProperties({
+				label: 'Copy-only backup'
+			})
+			.component();
+			checkbox.onChanged(e => {
+						console.info("inputBox.enabled " + inputBox.enabled);
+						inputBox.enabled = !inputBox.enabled;
+			});
+			let button = view.modelBuilder.button()
+			.withProperties({
+				label: '+'
+			}).component();
+			let button3 = view.modelBuilder.button()
+			.withProperties({
+			label: '-'
+
+			}).component();
+			let button2 = view.modelBuilder.button()
+			.component();
+			button.onDidClick(e => {
+							inputBox2.value = 'Button clicked';
+			});
+			let dropdown = view.modelBuilder.dropDown()
+			.withProperties({
+				value: 'Full',
+				values: ['Full', 'Differential', 'Transaction Log']
+			})
+			.component();
+			let f = 0;
+			inputBox.onTextChanged((params) => {
+			vscode.window.showInformationMessage(inputBox.value);
+				f = f + 1;
+				inputBox2.value=f.toString();
+			});
+			dropdown.onValueChanged((params) => {
+					vscode.window.showInformationMessage(inputBox2.value);
+					inputBox.value = dropdown.value;
+			});
+			let formModel = view.modelBuilder.formContainer()
+			.withFormItems([{
+						component: inputBox,
+						title: 'Backup name'
+				}, {
+						component: inputBox2,
+						title: 'Recovery model'
+				}, {
+						component:dropdown,
+						title: 'Backup type'
+				}, {
+					component: checkbox,
```

Comment:
Why no title? This looked weird to me without a title or label. Speaking of, would be nice to set label during this method for checkboxes :)

Reply:
usually check box in the form is assigned to checkbox itself

---
In reply to: [188370922](https://github.com/Microsoft/sqlopsstudio/pull/1415#discussion_r188370922) [](ancestors = 188370922)